packages/table/src/common/utils.ts

```diff
@@ -333,4 +344,34 @@ export const Utils = {
     isLeftClick(event: MouseEvent) {
         return event.button === 0;
     },
+
+    /**
+     * Returns true if the arrays are equal. Elements will be shallowly compared by default, or they
+     * will be compared using the custom `compare` function if one is provided.
+     */
+    arraysEqual(arrA: any[], arrB: any[], compare = (a: any, b: any) => a === b) {
+        // treat `null` and `undefined` as the same
+        if (arrA == null && arrB == null) {
+            return true;
+        } else if (arrA == null || arrB == null || arrA.length !== arrB.length) {
+            return false;
+        } else {
+            return arrA.every((a, i) => compare(a, arrB[i]));
+        }
+    },
 };
+
+/**
+ * Partial shallow comparison between objects using the given list of keys.
+ */
+function _shallowCompareKeys(objA: any, objB: any, keys: string[]) {
```

Comment:
Would literally be the second time I've actually had to use recursion in the wild, ha. Let me dust off my knowledge and give it a try.